nixos/modules/system/boot/unl0kr.nix

```diff
@@ -0,0 +1,93 @@
+{ config, lib, pkgs, utils, ... }:
+
+with lib;
+
+let
+  cfg = config.boot.initrd.unl0kr;
+in
+{
+  options.boot.initrd.unl0kr = {
+    enable = mkEnableOption (lib.mdDoc "unl0kr in initrd") // {
+      description = lib.mdDoc ''
+        Whether to enable the unl0kr on-screen keyboard in initrd to unlock LUKS.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    meta.maintainers = with maintainers; [ tomfitzhenry ];
+    assertions = [
+      {
+        assertion = cfg.enable -> config.boot.initrd.systemd.enable;
+        message = "boot.initrd.unl0kr is only supported with boot.initrd.systemd.";
+      }
+    ];
+    boot.initrd.availableKernelModules = [
+      "evdev" # for entering pw
+      "bochs"
+    ];
+
+    boot.initrd.systemd = {
+      storePaths = with pkgs; [
+        gnugrep
+        libinput
+        xkeyboard_config
+        systemd
+        unl0kr
+      ];
```

Comment:
These should probably be minimized, if possible. Like ideally we would be overriding `unl0kr` or whatever needs `systemd` to use `config.boot.initrd.systemd.package` instead. And only the files specifically needed from these other packages would be listed here.

Reply:
@tomfitzhenry I still think these need to be reduced further. As is, you're including things like manpages. Like maybe just the `/bin` subdirectories or something like that.